src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -980,7 +980,20 @@ private static int CanonicallyCompareDiagnostics(Diagnostic x, Diagnostic y)
             }
 
             // Optimization: don't bother 
-            return codeCompare == 0 ? string.CompareOrdinal(x.ToString(), y.ToString()) : codeCompare;
+            if (codeCompare != 0)
+                return codeCompare;
+
+            for (int i = 0; i < x.Arguments.Count && i < y.Arguments.Count; i++)
```

Comment:
>x.Arguments.Count && i < y.Arguments.Count [](start = 32, length = 42)

Consider extracting `int n = Math.Min(x.Arguments.Count, y.Arguments.Count);` and comparing `i < n`.

Reply:
Sure, but why? Code style or performance?

Reply:
Code style. It's a minor point only.

---
In reply to: [263482540](https://github.com/dotnet/roslyn/pull/33930#discussion_r263482540) [](ancestors = 263482540)